include/tsid/math/fwd.hpp

```diff
@@ -20,8 +20,13 @@
 
 #include <Eigen/Core>
 
-#define EIGEN_MALLOC_ALLOWED Eigen::internal::set_is_malloc_allowed(true);
-#define EIGEN_MALLOC_NOT_ALLOWED Eigen::internal::set_is_malloc_allowed(false);
+#ifdef EIGEN_RUNTIME_NO_MALLOC
```

Comment:
EIGEN_MALLOC_ALLOWED is only for exceptions when EIGEN_RUNTIME_NO_MALLOC is set.
This issue has been solved in Eigen 3.3.x